File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/PublisherManga.java
package com.amakakeru.mangaworld.Entity;

import com.amakakeru.mangaworld.Entity.IdClasses.PublisherMangaId;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Objects;

@Data
@Entity
@IdClass(PublisherMangaId.class)
@Table(name = "publisher_manga")
public class PublisherManga {

    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "p_id")
    private Long pid;

    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "m_id")
    private Long mid;

    @ManyToOne
    @JoinColumn(name = "p_id", insertable = false, updatable = false)
    private Publisher publisher;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;

    @Override
    public int hashCode() {
        return Objects.hash(pid, mid);
    }

    @Override
    public String toString() {
        return "PublisherManga{" +
                "pid=" + pid +
                ", mid=" + mid +
                '}';
    }
}

File: Frontend/manga-world/src/Model/PublisherManga.ts
import Manga from "./Manga";
import Publisher from "./Publisher";

class PublisherManga {
  pid: number;
  mid: number;
  publisher: Publisher;
  manga: Manga;

  constructor(pid: number, mid: number, publisher: Publisher, manga: Manga) {
    this.pid = pid;
    this.mid = mid;
    this.publisher = publisher;
    this.manga = manga;
  }
}

export default PublisherManga;
